### PR TITLE
Improve obsolete message for SetMessageHeader

### LINKED ISSUE #3725: SetMessageHeader Obsolete message is not very clear on what needs to be done.
The current obsolete message is here:
https://github.com/Particular/NServiceBus/blob/7df2cc7f289629072b1a13c8adead40efe0eb180/src/NServiceBus.Core/obsoletes.cs#L689

The upgrade guide section documents this behavior:
http://docs.particular.net/nservicebus/upgrades/5to6#header-management-setting-headers-on-outgoing-messages
Talks about adding header to all outgoing messages, but doesn't highlight how to set a specific header. 

Relates to: https://github.com/Particular/V6Launch/issues/35

### PR DESCRIPTION
**Pull request by timbussmann**
fixes #3725

@Particular/nservicebus-maintainers @indualagarsamy ping

### CODE REVIEW

**Comment by andreasohlund** [comment]
LGTM, please merge if you're good with the wording @indualagarsamy

File: src/NServiceBus.Core/obsoletes.cs [hunk 1]
```diff
@@ -1697,7 +1697,7 @@ public class TransactionSettings
             TreatAsErrorFromVersion = "6.0",
             Message =
                 @"DoNotWrapHandlersExecutionInATransactionScope() has been removed since transaction scopes are no longer used by non DTC transports delay the dispatch of all outgoing operations until handlers have been executed.
-In Version 6 handlers will only be wrapped in a transactionscope if running the MSMQ or SQLServer transports in default mode. This means that performing storage operations against data sources also supporting transaction scopes 
+In Version 6 handlers will only be wrapped in a transactionscope if running the MSMQ or SQLServer transports in default mode. This means that performing storage operations against data sources also supporting transaction scopes
```
Comment (danielmarbach) [inline src/NServiceBus.Core/obsoletes.cs]: This change is only whitespace, correct?

**Comment by danielmarbach** [comment]
:+1:

File: src/NServiceBus.Core/obsoletes.cs (hunk 1 above)
Comment (timbussmann) [inline src/NServiceBus.Core/obsoletes.cs]: yes

**Comment by danielmarbach** [comment]
ping @indualagarsamy

File: src/NServiceBus.Core/obsoletes.cs [hunk 2]
```diff
@@ -686,7 +686,7 @@ public static string GetMessageHeader(this IBus bus, object msg, string key)
         [ObsoleteEx(
             RemoveInVersion = "7.0",
             TreatAsErrorFromVersion = "6.0",
-            Message = "Headers can be set using the ``.SetHeader` method on the context object passed into the behavior or mutator")]
+            Message = @"Use the overload of the Send, Publish or Reply method that accepts an options parameter. Call options.SetHeader(""MyHeader"",""MyValue"") instead.")]
```
Comment (indualagarsamy) [inline src/NServiceBus.Core/obsoletes.cs]: @timbussmann - What does `The member currently throws a NotImplementedException.` mean? Which member throws an Not implemented exception?

Comment (bording) [inline src/NServiceBus.Core/obsoletes.cs]: @indualagarsamy That line in the obsolete warning from the `ObsoleteEx` fody attribute, and it's referring to the implemenation stub left around for the warning. See https://github.com/Particular/NServiceBus/pull/3784/files#diff-287c0fdac06c80a87ce22cb7ee81ac17R692

Comment (SimonCropp) [inline src/NServiceBus.Core/obsoletes.cs]: https://github.com/Fody/Obsolete/blob/master/README.md#throwsnotimplementedtext

**Pull request merged**